src/ext/java/org/rutebanken/tiamat/ext/fintraffic/auth/FintrafficAuthorizationService.java

```diff
@@ -0,0 +1,155 @@
+
+    private static final Set<SubmodeEnumuration> ALL_SUBMODES = EnumSet.allOf(SubmodeEnumuration.class);
+
+    private final Logger logger = LoggerFactory.getLogger(FintrafficAuthorizationService.class);
+
+    private final TrivoreAuthorizations trivoreAuthorizations;
+
+    public FintrafficAuthorizationService(TrivoreAuthorizations trivoreAuthorizations) {
+        this.trivoreAuthorizations = trivoreAuthorizations;
+    }
+
+    @Override
+    public boolean canEditAllEntities() {
+        return trivoreAuthorizations.hasAccess(ENTITY_TYPE_ALL, TRANSPORT_MODE_ALL, MANAGE);
+    }
+
+    @Override
+    public boolean canEditEntities(Collection<? extends EntityStructure> entities) {
+        return trivoreAuthorizations.hasAccess(ENTITY_TYPE_ALL, TRANSPORT_MODE_ALL, MANAGE)
+                || entities.stream().allMatch(this::canEditEntity);
+    }
+
+    @Override
+    public void verifyCanEditEntities(Collection<? extends EntityStructure> entities) {
+        if (!canEditEntities(entities)) {
+            throw new AccessDeniedException("current user is not allowed to edit entities");
+        }
+    }
+
+    @Override
+    public void verifyCanDeleteEntities(Collection<? extends EntityStructure> entities) {
+        if (!entities.stream().allMatch(this::canDeleteEntity)) {
+            throw new AccessDeniedException("current user is not allowed to delete entities");
+        }
+    }
+
+    @Override
+    public boolean canDeleteEntity(EntityStructure entity) {
+        String codespace = getCodespace(entity.getNetexId());
+        return trivoreAuthorizations.hasAccess(detectEntityType(entity), detectTransportMode(entity), ADMINISTER)
+                && trivoreAuthorizations.hasAccessToCodespace(codespace);
+    }
+
+    @Override
+    public boolean canEditEntity(EntityStructure entity) {
+        String codespace = getCodespace(entity.getNetexId());
+        return trivoreAuthorizations.hasAccess(detectEntityType(entity), detectTransportMode(entity), MANAGE)
+                && trivoreAuthorizations.hasAccessToCodespace(codespace);
+    }
+
+    private static String detectEntityType(EntityStructure entity) {
+        return entity.getClass().getSimpleName();
+    }
+
+    private static String detectTransportMode(EntityStructure entity) {
+        return switch (entity) {
+            case StopPlace sp -> sp.getTransportMode().value();
+            default -> TRANSPORT_MODE_ALL;
+        };
+    }
+
+    @Override
+    public boolean canEditEntity(Point point) {
+        logger.trace("FintrafficAuthorizationService.canEditEntity({})", point);
+        return true;  // TODO: implement geofencing support
+    }
+
+    @Override
+    public Set<StopTypeEnumeration> getAllowedStopPlaceTypes(EntityStructure entity) {
+        logger.trace("FintrafficAuthorizationService.getAllowedStopPlaceTypes({}, {})", entity.getId(), entity.getNetexId());
```

Comment:
All methods which don't "do" anything besides returning a value have a trace logging paired with them just to see that they're called. In our scheme we're currently allowing everything everywhere, this probably should be done in some other way but as it wasn't very obvious what the value(s) here should be, they've been left as-is for now.

For example, if `EntityStructure´ is a `StopPlace`, it has `StopTypeEnumeration` field. Am I supposed to detect and return that here, or is there an implicit hierarchy of various `StopTypeEnumeration` values somewhere I could reuse to return sensible defaults?